### PR TITLE
Removed custom defaults dialog from timeout

### PR DESCRIPTION
**Pull request by Asizon**
The currently used timeout for reset to custom defaults window only offers 10 seconds, It is not enough to read the content of the window and proceed. increased timeout to 15 seconds for this case.

### CODE REVIEW

**Comment by mikeller** [comment]
Not sure what flow the timeout that you are changing is in - can you please explain what steps lead to this timeout, or even show a video?

**Comment by Asizon** [comment]
I'm really not sure if this function needs timeout. It can be reproduced after flashing, when opens the window to apply custom defaults, you have 10 seconds before disconnecting the port by the timeout, and it turn closing the window because now we have configured in this way, to close when port is disconected

**Comment by mikeller** [comment]
@Asizon: Yes, that is entirely unintentional, and caused by the connection not marked as 'successful' at the point that this dialog pops up. The configuration warning that pops up only a couple of steps later is not subject to this problem.

A better solution will be to just move the connection acknowledgement that is needed to disable the timer to before the custom defaults warning - if we get to this point the flight controller is responding, so we don't really need the timeout to still be active.

**Comment by Asizon** [comment]
Ok @mikeller , I think now it is done in the correct way,thanks. Moved timeout killer before custom defaults dialog.

File: src/js/serial_backend.js [hunk 1]
```diff
@@ -466,14 +464,12 @@ function onConnect() {
         $('div#flashbutton a.flash_state').removeClass('active');
         $('div#flashbutton a.flash').removeClass('active');
     }
-    GUI.timeout_remove('connecting'); // kill connecting timer
```
Comment (mikeller) [inline src/js/serial_backend.js:469]: Why are you removing this? This is called _after_ the problem report dialog, so I don't think that this is the problem.

File: src/js/serial_backend.js [hunk 2]
```diff
@@ -338,10 +340,6 @@ function processBoardInfo() {
             MSP.send_message(MSPCodes.MSP_RESET_CONF, buffer, false);
 
             dialog.close();
-
-            GUI.timeout_add('connecting', function () {
```
Comment (mikeller) [inline src/js/serial_backend.js:342]: Why are you removing this?

Comment (Asizon) [reply src/js/serial_backend.js]: ups i have removed validation timeout,fixed now. Also readded old timeout killer,I thought that having the other one, we wouldn't need it.

**Comment by mikeller** [comment]
I think before we propose any changes here we need to understand _why_ this is happening - from looking at the code it looks like the timeout should either apply to both the custom defaults dialog and the problem dialog, or to none, but not to only one of them.

**Comment by Asizon** [comment]
Ok @mikeller , now i only define timeout killer if custom defaults and reportproblems dialogs are executed

**Comment by Asizon** [comment]
Closing it, thanks for the new solution @mikeller

**Pull request closed without merging**